dart/lib/src/sentry_client.dart

```diff
@@ -379,6 +415,21 @@ class SentryClient {
     return id ?? SentryId.empty();
   }
 
+  bool _isIgnoredTransaction(SentryTransaction transaction) {
+    if (_options.ignoreTransactions == null ||
+        _options.ignoreTransactions!.isEmpty) {
+      return false;
+    }
+
+    String combinedRegexPattern = _options.ignoreTransactions!.join('|');
+
+    RegExp regExp = RegExp(combinedRegexPattern);
+
+    bool ignore = regExp.firstMatch(transaction.tracer.name) != null;
```

Comment:
```suggestion
    final combinedRegexPattern = _options.ignoreTransactions!.join('|');

    final regExp = RegExp(combinedRegexPattern);

    final ignore = regExp.firstMatch(transaction.tracer.name) != null;
```

let's keep immutability as default, same for `_isIgnoredError`